### PR TITLE
Issue #2894020 by WidgetsBurritos: Run Storage Entities

### PR DESCRIPTION
**Pull request by WidgetsBurritos**
Drupal Issue: https://www.drupal.org/node/2894020

This PR adds run storage content entities.  

 - 581bb25 - Generate run entities via Drupal Console
 - 76bacc5 - Reorganize file structure
 - 2c22738 - Change URL paths and remove revert features
 - 3407b8d - Connect config and content entities
 - b725d3e - Generate capture field type/formatter/widget via Drupal Console
 - f3e165d - Add capture run field
 - f7e5983 - Fix code standards
 - 894ae29 - Add functional test coverage for run entity creation.
 - 11f0ade - Namespace the field type.
 - 6278d34 - Fixes duplicate content bug.
 - 1af9116 - Fixes borked test.

### CODE REVIEW

File: src/Controller/WebPageArchiveRunController.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    $rows = [];
+
+    $vids = $web_page_archive_run_storage->revisionIds($web_page_archive_run);
+
+    $latest_revision = TRUE;
+
+    foreach (array_reverse($vids) as $vid) {
+      /** @var \Drupal\web_page_archive\WebPageArchiveRunInterface $revision */
+      $revision = $web_page_archive_run_storage->loadRevision($vid);
+      // Only show revisions that are affected by the language that is being
```
Comment (bighappyface) [inline src/Controller/WebPageArchiveRunController.php:81]: Artifact comment?

Comment (WidgetsBurritos) [reply src/Controller/WebPageArchiveRunController.php:81]: No, it just spanned two lines.  That said, it was auto generated by drupal console, but unless we decide to remove this code later, I'd rather not remove potentially helpful comments.   Thoughts?

Comment (bighappyface) [reply src/Controller/WebPageArchiveRunController.php:81]: I meant that the comment was an artifact from elsewhere, such as Drupal Console. The reference to languages for untranslatable entities is funny, but not an issue for me.

Comment (WidgetsBurritos) [reply src/Controller/WebPageArchiveRunController.php:81]: Ahh got it. I'm sure there's a bit of drupal consoleness in here.   I can either remove it now, or we can work on cleaning some of that stuff later.  Lemme know how you wanna tackle it.

Comment (bighappyface) [reply src/Controller/WebPageArchiveRunController.php:81]: We are good here. Bikeshedding averted...

**Review by bighappyface — APPROVED** [review]
+1 for 581bb25
+1 for 76bacc5
+1 for 2c22738
+1 for 3407b8d
+1 for b725d3e
+1 for f3e165d
+1 for f7e5983
+1 for 894ae29
+1 for 11f0ade
+1 for 6278d34
+1 for 1af9116

File: src/Entity/WebPageArchive.php [hunk 2]
```diff
@@ -294,16 +295,25 @@ public function startNewRun(HandlerStack $handler = NULL) {
...
   /**
-   * Initializes and retrieves run entity.
+   * Retrieves the run entity for this config entity.
+   *
+   * @return \Drupal\web_page_archive\Entity\WebPageArchiveRun
    */
-  protected function initializeRunEntity() {
+  public function getRunEntity() {
     $entity = NULL;
     if (isset($this->entity_run)) {
```
Comment (writes) [inline src/Entity/WebPageArchive.php]: Always run?

**Comment by writes** [comment]
LGTM!

Nice work @WidgetsBurritos!

+1 for 6abe3fd
+1 for 743ed74
+1 for b570e87
+1 for 0b56b89
+1 for 4e49cac
+1 for 20a9bc3
+1 for 974a3e7
+1 for 5a98064
+1 for 521cd60
+1 for 6278d34

**Review by writes — APPROVED** [review]

**Comment by writes** [comment]
+1 1af9116

**Review by writes — APPROVED** [review]

**Review by writes — APPROVED** [review]

**Pull request merged**